language-scripts/nodejs/dev-assemble

```diff
@@ -1,16 +1,66 @@
 #!/bin/bash
+
 set -e
 set -x
 
-cd /tmp/src
+# DEBUG_MODE is true by default, which means that the application will be started with remote debugging enabled.
+DEBUG_MODE="${DEBUG_MODE:=true}"
 
-NODE_ENV="${NODE_ENV:=development}"
+if [ "$DEBUG_MODE" == "true" ]; then
+	export DEV_MODE=true
+fi
 
 REFRESH_WORKING_DIR=/tmp/refresh/
 PACKAGE_SHA_FILE=$REFRESH_WORKING_DIR/package.json.sha1
-
 mkdir -p $REFRESH_WORKING_DIR
 
+#from: /usr/libexec/s2i/env
+# If NODE_ENV is not set by the user, then default to production.
+# User may also set DEV_MODE=true
+if [ -z "$NODE_ENV" ]; then
+  if [ "$DEV_MODE" == true ]; then
+    export NODE_ENV=development
+  else
+    export NODE_ENV=production
+  fi
+fi
+
+# Set the environment for this build configuration to production by default.
+if [ "$NODE_ENV" == "production" ]; then
+  export DEV_MODE=false
+else
+  export DEV_MODE=true
+fi
+
+
+if [ "$DEV_MODE" == true ] ; then
+	set -x
+fi
+
+echo "---> Installing application source"
+# Do not run it with -p (preserving  some attributes)
+# It causes it to fail with "cp: preserving times for './.': Operation not permitted" when files come from Windows
+#cp -Rfp /tmp/src/. ./
+cp -Rf /tmp/src/. ./
+
+
+
+# don't do anything if package.json is the same as the last time this script was executed
+if [ "$DEV_MODE" == true  ] && [ -f $PACKAGE_SHA_FILE ]; then
+    out=$(sha1sum -c $PACKAGE_SHA_FILE || true)
+    if [ "$out" == "package.json: OK" ]; then
+        echo "---> package.json is the same as last time, no changes needed."
+		echo "---> Fix permissions on app-root"
+		fix-permissions /opt/app-root
+        exit 0
+    fi
+fi
+
+echo "---> package.json modified"
```

Comment:
This would get executed even if the package.json was not modified, no? Or does `fix-permissions` have the provision to exit after doing `fix-permissions /opt/app-root` a few steps above?

Reply:
there is `exit 0` right after `fix-permissions` on the line `55`

There is wrong formatting, that is making harder to see it.